--- a/util/seqset.v
+++ b/util/seqset.v
@@ -21,7 +21,7 @@
   HB.instance Definition _  setSubType := [isSub for _set_seq].
   HB.instance Definition _ := [Equality of set by <:].
   Canonical Structure mem_set_predType := PredType (fun (l : set) => mem_seq (_set_seq l)).
-  Definition set_of of phant T := set.
+  Definition set_of (_ : phant T) := set.
 
 End SeqSet.
 Set Warnings "redundant-canonical-projection".
--- a/util/seqset.v
+++ b/util/seqset.v
@@ -24 +24 @@
-  Definition set_of of phant T := set.
+  Definition set_of (_ : phant T) := set.
