--- a/implementation/definitions/extrapolated_arrival_curve.v
+++ b/implementation/definitions/extrapolated_arrival_curve.v
@@ -1,5 +1,7 @@
 Require Export prosa.behavior.time.
 Require Export prosa.util.all.
+From mathcomp Require Export div path.
+Require Export prosa.util.epsilon.
 
 (** This file introduces an implementation of arrival curves via the
     periodic extension of finite _arrival-curve prefix_. An
