--- a/implementation/refinements/arrival_bound.v
+++ b/implementation/refinements/arrival_bound.v
@@ -1,5 +1,6 @@
 Require Export prosa.implementation.refinements.refinements.
 Require Export prosa.implementation.definitions.arrival_bound.
+Set SsrOldRewriteGoalsOrder.
 
 (** In this section, we define a generic version of arrival bound and
     prove that it is isomorphic to the standard, natural-number-based
