--- a/implementation/facts/extrapolated_arrival_curve.v
+++ b/implementation/facts/extrapolated_arrival_curve.v
@@ -1,6 +1,7 @@
 Require Export prosa.behavior.time.
 Require Export prosa.util.all.
 Require Export prosa.implementation.definitions.extrapolated_arrival_curve.
+Set SsrOldRewriteGoalsOrder.
 
 (** In this file, we prove basic properties of the arrival-curve
     prefix and the [extrapolated_arrival_curve] function. *)
