--- a/implementation/refinements/EDF/fast_search_space.v
+++ b/implementation/refinements/EDF/fast_search_space.v
@@ -1,5 +1,6 @@
 Require Export prosa.results.rta.ideal.edf.bounded_nps.
 Require Export prosa.implementation.refinements.fast_search_space_computation.
+Set SsrOldRewriteGoalsOrder.
 
 (** First, we define the concept of higher-or-equal-priority workload.  *)
 Definition bound_on_total_hep_workload (ts : seq Task) (tsk : Task) A Δ :=
