diff --git a/analysis/definitions/sbf/pred.v b/analysis/definitions/sbf/pred.v
--- a/analysis/definitions/sbf/pred.v
+++ b/analysis/definitions/sbf/pred.v
@@ -2,6 +2,8 @@
 Require Export prosa.behavior.schedule.
 Require Export prosa.model.processor.supply.
 Require Export prosa.analysis.definitions.sbf.sbf.
+Require Import prosa.util.rel.
+Require Import prosa.util.setoid.
 
 (** ** SBF Parameter Semantics *)
