--- a/model/processor/multiprocessor.v
+++ b/model/processor/multiprocessor.v
@@ -1,5 +1,4 @@
 Require Export prosa.behavior.all.
-Require Import prosa.analysis.facts.behavior.service.
 
 (** * Multiprocessor State *)
 
@@ -73,9 +72,16 @@
       service_on := multiproc_service_on
     |}.
   Next Obligation.
-    by move => ? ? ?; apply: leq_sum => c _; exact: service_on_le_supply_on.
+    rewrite /multiproc_service_on /multiproc_supply_on /service_in /supply_in.
+    by apply: leq_sum => c _; exact: service_on_le_supply_on.
   Qed.
-  Next Obligation. by move=> ? ? ?; apply: service_in_implies_scheduled_in. Qed.
+  Next Obligation.
+    match goal with H : is_true (~~ _) |- _ => move: H end.
+    rewrite /multiproc_scheduled_on /multiproc_service_on /scheduled_in /service_in.
+    move=> /existsP Hsched; apply/eqP; rewrite sum_nat_eq0; apply/forallP => c.
+    rewrite service_on_implies_scheduled_on//.
+    by apply/negP => ?; apply: Hsched; exists c.
+  Qed.
 
   (** From the instance [multiproc_state], we get the function [service_in].
       The service received by a given job [j] in a given multiprocessor state
